crates/rpc/rpc-types/src/eth/transaction/mod.rs

```diff
@@ -71,6 +71,20 @@ pub struct Transaction {
     /// Some(1) for AccessList transaction, None for Legacy
     #[serde(rename = "type", skip_serializing_if = "Option::is_none")]
     pub transaction_type: Option<U64>,
+
+    /// Hash that uniquely identifies the source of the deposit.
+    #[cfg(feature = "optimism")]
+    #[serde(rename = "sourceHash", skip_serializing_if = "Option::is_none")]
+    pub source_hash: Option<B256>,
+    /// The ETH value to mint on L2
+    #[cfg(feature = "optimism")]
+    #[serde(rename = "mint", skip_serializing_if = "Option::is_none")]
+    pub mint: Option<U128>,
+    /// Field indicating whether the transaction is a system transaction, and therefore
+    /// exempt from the L2 gas limit.
+    #[cfg(feature = "optimism")]
+    #[serde(rename = "isSystemTx", skip_serializing_if = "Option::is_none")]
+    pub is_system_tx: Option<bool>,
```

Comment:
I'd also like this as a separate container that uses flatten

Reply:
addressed in https://github.com/anton-rs/op-reth/pull/133